File: internal/content/domain/media.go
package domain

import (
	"strings"
	"time"

	"github.com/google/uuid"
)

type MediaType string

const (
	MediaImage MediaType = "image"
	MediaVideo MediaType = "video"
)

func ValidMediaType(t MediaType) bool {
	return t == MediaImage || t == MediaVideo
}

type Media struct {
	ID        uuid.UUID
	ContentID uuid.UUID
	URL       string
	MediaType MediaType
	Position  int
	CreatedAt time.Time
}

func NewMedia(contentID uuid.UUID, url, mediaType string, position int) (Media, error) {
	url = strings.TrimSpace(url)
	if url == "" {
		return Media{}, ErrMediaURLRequired
	}

	mediaValue := MediaType(defaultString(strings.TrimSpace(strings.ToLower(mediaType)), string(MediaImage)))
	if !ValidMediaType(mediaValue) {
		return Media{}, ErrInvalidMediaType
	}

	if position < 0 {
		position = 0
	}

	return Media{
		ID:        uuid.New(),
		ContentID: contentID,
		URL:       url,
		MediaType: mediaValue,
		Position:  position,
	}, nil
}

File: frontend/apps/dashboard/features/content/types/media.ts
export type MediaType = "image" | "video";

export interface Media {
  id: string;
  contentId: string;
  url: string;
  mediaType: MediaType;
  position: number;
  createdAt: string;
}

export interface MediaDto {
  id: string;
  content_id: string;
  url: string;
  media_type: string;
  position: number;
  created_at: string;
}

export function toMedia(dto: MediaDto): Media {
  return {
    id: dto.id,
    contentId: dto.content_id,
    url: dto.url,
    mediaType: dto.media_type === "video" ? "video" : "image",
    position: dto.position ?? 0,
    createdAt: dto.created_at,
  };
}

export interface AttachMediaInput {
  url: string;
  mediaType: MediaType;
  position?: number;
}
